packages/server/src/agents.storage.ts

```diff
@@ -0,0 +1,410 @@
+import { Injectable } from '@nestjs/common';
+import { ConfigurationService } from '../config/configuration.js';
+import {
+  AgentMode,
+  AgentSystem,
+  AgentSystemConfig,
+  createContextFromJson,
+} from '@snakagent/agents';
+import { Postgres } from '@snakagent/database';
+import { SystemMessage } from '@langchain/core/messages';
+import { AgentPromptSQL, AgentConfigSQL } from './interfaces/sql_interfaces.js';
+import {
+  logger,
+  AgentConfig,
+  ModelsConfig,
+  ModelLevelConfig,
+  ModelProviders,
+} from '@snakagent/core';
+
+export interface AgentConfigJson {
+  name: string;
+  prompt: AgentPromptSQL;
+  interval: number;
+  plugins: string[];
+  memory: {
+    enabled: boolean;
+    short_term_memory_size: number;
+  };
+}
+
+@Injectable()
+export class AgentStorage {
+  private agentConfigs: AgentConfigSQL[] = [];
+  private agentInstances: Map<number, AgentSystem> = new Map();
+  private initialized: boolean = false;
+  private initPromise: Promise<void>;
+  constructor(private readonly config: ConfigurationService) {
+    this.initPromise = this.initialize();
+  }
+
+  private async initalize_database() {
+    try {
+      const q = [
+        new Postgres.Query(`
+          CREATE TYPE agent_prompt AS (
+            bio TEXT,
+            lore TEXT[] ,
+            objectives TEXT[],
+            knowledge TEXT[]
+          );`),
+        new Postgres.Query(`
+          CREATE TYPE memory AS (
+              memory BOOLEAN,
+              shortTermMemorySize INTEGER
+          );`),
+        new Postgres.Query(`
+          CREATE TYPE model AS (
+              provider TEXT,
+              model_name TEXT,
+              description TEXT
+          );`),
+        // TODO : Use id as the unique
+        new Postgres.Query(`
+            CREATE TABLE IF NOT EXISTS agents (
+            id SERIAL PRIMARY KEY,
+            name VARCHAR(255) NOT NULL,
+            group_id INTEGER DEFAULT 0,
+            prompt agent_prompt NOT NULL,
+            interval INTEGER NOT NULL DEFAULT 5,
+            plugins TEXT[] NOT NULL DEFAULT '{}',
+            memory memory NOT NULL DEFAULT (false, 5)
+            );`),
+        new Postgres.Query(`
+            CREATE TABLE IF NOT EXISTS agent_iterations (
+            id SERIAL PRIMARY KEY,
+            data JSONB NOT NULL
+            );`),
+        new Postgres.Query(`
+            CREATE TABLE IF NOT EXISTS conversation (
+            conversation_id SERIAL PRIMARY KEY,
```

Comment:
conversation_id is TEXT in the message table but SERIAL in the conversation table